### PR TITLE
Week 2 Assignments

### PR DESCRIPTION
**Pull request by pradyum98**
Assignments for Week 2

### CODE REVIEW

File: Week-2/Session-4/BirthdayMessageCreator.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Birthday Message Creator</title>
+    <style>
+        body {
+            position: relative;
+        }
+        /* Why do i have to define this ?
+        Without this css block my absolute positioning doesn't work */
```
Comment (kamalmohit) [inline Week-2/Session-4/BirthdayMessageCreator.html:13]: Absolute positioning always works based off the closest relative based parent. The why is this necessary is just how the CSS functions.

File: Week-2/Session-4/SocialNetworkForm.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            display: flex;
+            margin-bottom: 10px;
+        }
+         .textarea-row {
+            display: flex;
+            margin-bottom: 45px;
+        }
+        input {
+            position: absolute;
+            margin-left: 150px;
```
Comment (kamalmohit) [inline Week-2/Session-4/SocialNetworkForm.html:24]: Same. Try using the gap property to space out flex child elements. This is a much more responsive behaviour.

**Review by kamalmohit — CHANGES_REQUESTED** [review]
https://css-tricks.com/snippets/css/a-guide-to-flexbox/

Please take a look at the comments I have added to improve the CSS code. You can refer to the above link to understand more about the flex behaviour.

File: Week-2/Session-4/BirthdayMessageCreator.html (hunk 1 above)
Comment (pradyum98) [reply Week-2/Session-4/BirthdayMessageCreator.html:13]: Noted

File: Week-2/Session-6/Recipe/index.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      <div class="content">
+         <div class="main-content">
+            <div class="title">
+               <h1>OATMEAL COCONUT PANTRY COOKIES</h1>
+               <p>Chef Roman, August 24, 2018</p>
+            </div>
+            <div class="summary">
+               <img src="oatmealCookie_small.jpg" alt="Photo of two of the world's best oatmeal cookies">
+               <div>
+                  <div>
```
Comment (kamalmohit) [inline Week-2/Session-6/Recipe/index.html:26]: We can use a more semantic approach here and instead of using divs, we can also use sections and articles HTML tag to make it more semantic.

**Review by kamalmohit — CHANGES_REQUESTED** [review]
Please see the comment mentioned @pradyum98 
Rest everything is fine!

**Comment by pradyum98** [comment]
> Please see the comment mentioned @pradyum98 Rest everything is fine!

Thanks @kamalmohit 
Will address the comments and push required changes

File: Week-2/Session-6/Recipe/index.html (hunk 3 above)
Comment (pradyum98) [reply Week-2/Session-6/Recipe/index.html:26]: hey @kamalmohit 
This was part of the boilerplate code provided hence didn't modify it

File: Week-2/Session-4/BirthdayMessageCreator.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        .input-row {
+            display: flex;
+            margin-bottom: 10px;
```
[the remark this answers is not in the document]
Comment (pradyum98) [reply Week-2/Session-4/BirthdayMessageCreator.html]: Understood
Changes have been made

File: Week-2/Session-4/SocialNetworkForm.html (hunk 2 above)
Comment (pradyum98) [reply Week-2/Session-4/SocialNetworkForm.html]: Done

File: Week-2/Session-4/BirthdayMessageCreator.html [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        }
+         .textarea-row {
+            display: flex;
+            margin-bottom: 45px;
+        }
+        input {
+            position: absolute;
+            margin-left: 150px;
```
[the remark this answers is not in the document]
Comment (pradyum98) [reply Week-2/Session-4/BirthdayMessageCreator.html]: Done
